crd/apis/network/v1/gkenetworkparams_types.go

```diff
@@ -0,0 +1,73 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster
+// +kubebuilder:storageversion
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// GKENetworkParams represent GKE specific parameters for the network.
+type GKENetworkParams struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   GKENetworkParamsSpec   `json:"spec,omitempty"`
+	Status GKENetworkParamsStatus `json:"status,omitempty"`
+}
+
+// DeviceModeType defines mode in which the devices will be used by the Pod
+// +kubebuilder:validation:Enum=DPDK-UIO;DPDK-VFIO;NetDevice;RDMA
+type DeviceModeType string
+
+const (
+	// DpdkUIO indicates that NICs are bound to uio_pci_generic driver
+	DpdkUIO DeviceModeType = "DPDK-UIO"
+	// DpdkVFIO indicates that NICs are bound to vfio-pci driver
+	DpdkVFIO DeviceModeType = "DPDK-VFIO"
+	// NetDevice indicates that NICs are bound to kernel driver and used as net device
+	NetDevice DeviceModeType = "NetDevice"
```

Comment:
@liuyuan10 is the yaml case sensitive? Should we just do "netdevice" (or all CAPS) ?

Reply:
It is case sensitive. I used upper case because the exiting network type is L3/L2. I don't have strong preference but it seems normal to only capitalize the first char given k8s is very golang focused. For example, in anthos config, the lb kind is "kind: MetalLB"